clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignorTest.java

```diff
@@ -598,6 +555,43 @@ public void testLargeAssignmentAndGroupWithUniformSubscription() {
         assignor.assign(partitionsPerTopic, subscriptions);
     }
 
+    @Timeout(40)
+    @Test
+    public void testLargeAssignmentAndGroupWithNonEqualSubscription() {
+        // 1 million partitions!
+        int topicCount = 500;
+        int partitionCount = 2_000;
+        int consumerCount = 2_000;
+
+        List<String> topics = new ArrayList<>();
+        Map<String, Integer> partitionsPerTopic = new HashMap<>();
+        for (int i = 0; i < topicCount; i++) {
+            String topicName = getTopicName(i, topicCount);
+            topics.add(topicName);
+            partitionsPerTopic.put(topicName, partitionCount);
+        }
+        for (int i = 0; i < consumerCount; i++) {
+            if (i == consumerCount - 1) {
+                subscriptions.put(getConsumerName(i, consumerCount), new Subscription(topics.subList(0, 1)));
+            } else {
+                subscriptions.put(getConsumerName(i, consumerCount), new Subscription(topics));
+            }
+        }
+
+        Map<String, List<TopicPartition>> assignment = assignor.assign(partitionsPerTopic, subscriptions);
+
```

Comment:
Would it be worthwhile to add some verification after the `assign` calls in this test?

Reply:
It can, but this test: `testLargeAssignmentAndGroupWithNonEqualSubscription` is mainly to test **performance**, not functionality. We should cover the function testing in other tests. It's the same purpose for the above one: `testLargeAssignmentAndGroupWithUniformSubscription`.